vpr/src/route/rr_graph_reader.cpp

```diff
@@ -148,53 +150,170 @@ void load_rr_file(const t_graph_type graph_type,
-        process_edges(next_component, loc_data, wire_to_rr_ipin_switch, numSwitches);
+            process_nodes_and_switches_bin(fp, wire_to_rr_ipin_switch, is_global_graph, segment_inf, numSwitches);
+
+            partition_rr_graph_edges(device_ctx);
+            process_rr_node_indices(grid);
+            init_fan_in(device_ctx.rr_nodes, device_ctx.rr_nodes.size());
+            alloc_and_load_rr_indexed_data(segment_inf, device_ctx.rr_node_indices,
+                                           max_chan_width, *wire_to_rr_ipin_switch, base_cost_type);
+
+        } else {
+            /* Alloc rr nodes and count count nodes */
+            next_component = get_single_child(rr_graph, "rr_nodes", loc_data);
 
-        //Partition the rr graph edges for efficient access to configurable/non-configurable
-        //edge subsets. Must be done after RR switches have been allocated
-        partition_rr_graph_edges(device_ctx);
+            int num_rr_nodes = count_children(next_component, "node", loc_data);
 
-        process_rr_node_indices(grid);
+            device_ctx.rr_nodes.resize(num_rr_nodes);
+            device_ctx.connection_boxes.resize_nodes(num_rr_nodes);
+            process_nodes(next_component, loc_data);
 
-        init_fan_in(device_ctx.rr_nodes, device_ctx.rr_nodes.size());
+            next_component = get_single_child(rr_graph, "rr_edges", loc_data);
+            process_edges(next_component, loc_data, wire_to_rr_ipin_switch, numSwitches);
 
-        //sets the cost index and seg id information
-        next_component = get_single_child(rr_graph, "rr_nodes", loc_data);
-        set_cost_indices(next_component, loc_data, is_global_graph, segment_inf.size());
+            //Partition the rr graph edges for efficient access to configurable/non-configurable
+            //edge subsets. Must be done after RR switches have been allocated
+            partition_rr_graph_edges(device_ctx);
 
-        alloc_and_load_rr_indexed_data(segment_inf, device_ctx.rr_node_indices,
-                                       max_chan_width, *wire_to_rr_ipin_switch, base_cost_type);
+            process_rr_node_indices(grid);
 
-        process_seg_id(next_component, loc_data);
+            init_fan_in(device_ctx.rr_nodes, device_ctx.rr_nodes.size());
+
+            //sets the cost index and seg id information
+            next_component = get_single_child(rr_graph, "rr_nodes", loc_data);
+            set_cost_indices(next_component, loc_data, is_global_graph, segment_inf.size());
+
+            alloc_and_load_rr_indexed_data(segment_inf, device_ctx.rr_node_indices,
+                                           max_chan_width, *wire_to_rr_ipin_switch, base_cost_type);
+
+            process_seg_id(next_component, loc_data);
+        }
 
         device_ctx.chan_width = nodes_per_chan;
 
-        check_rr_graph(graph_type, grid, device_ctx.block_types);
+        //check_rr_graph(graph_type, grid, device_ctx.block_types);
```

Comment:
@andrewb1999  Just noticed this. Is it necessary to comment this check? If not it would be better to de-comment it.

Reply:
That was a mistake (although we are planning on adding an option to disable it eventually).  Should be fixed now.